src/main/java/capstone/mfslbackend/service/TransactionService.java

```diff
@@ -237,10 +269,129 @@ public Boolean approveTeam(Set<Player> players) {
+
+        }
+    }
+
+    private boolean checkExcedentPositions(FantasyTeam proposingFantasyTeam, String position) {
+
+        Long fantasyTeamId = proposingFantasyTeam.getId();
+
+        String[] playerIdsArray = proposingFantasyTeam.getPlayerIdsInOrder().split(" ");
+
+        List<Long> playerIdsList = Arrays.stream(playerIdsArray)
+                .map(Long::parseLong)
+                .toList();
+
+        long totalDefenders = proposingFantasyTeam.getPlayers().stream()
+                .filter(player -> player.getPosition().equals("Defender"))
+                .count();
+
+        long startingDefenders = playerIdsList.stream()
+                .limit(PLAYERS_IN_STARTING_LINEUP)
+                .map(playerService::getPlayerById)
+                .filter(player -> player != null && player.getPosition().equals("Defender"))
+                .count();
+
+        long totalMidfielders = proposingFantasyTeam.getPlayers().stream()
+                .filter(player -> player.getPosition().equals("Midfielder"))
+                .count();
+
+        long startingMidfielders = playerIdsList.stream()
+                .limit(PLAYERS_IN_STARTING_LINEUP)
+                .map(playerService::getPlayerById)
+                .filter(player -> player != null && player.getPosition().equals("Midfielder"))
+                .count();
+
+        long totalAttackers = proposingFantasyTeam.getPlayers().stream()
+                .filter(player -> player.getPosition().equals("Attacker"))
+                .count();
+
+        long startingAttackers = playerIdsList.stream()
+                .limit(PLAYERS_IN_STARTING_LINEUP)
+                .map(playerService::getPlayerById)
+                .filter(player -> player != null && player.getPosition().equals("Attacker"))
+                .count();
+
+
+        long startingGoalkeepers = playerIdsList.stream()
+                .limit(PLAYERS_IN_STARTING_LINEUP)
+                .map(playerService::getPlayerById)
+                .filter(player -> player != null && player.getPosition().equals("Goalkeeper"))
+                .count();
+
+        if (position.equals("Goalkeeper") && startingGoalkeepers >= MIN_GK) {
+            throw new Error400("Fantasy team with id " + fantasyTeamId + " cannot have more than 1 goalkeeper in the starting lineup");
+
+        } else if (!position.equals("Goalkeeper") && startingGoalkeepers < MIN_GK) { // might have to delete this condition
+            throw new Error400("Fantasy team with id " + fantasyTeamId + " needs at least 1 goalkeeper in the starting lineup");
+
+        } else if (position.equals("Defender")) {
+            if (totalDefenders > TOTAL_DEF) {
+                throw new Error400("Fantasy team with id " + fantasyTeamId + " cannot have more than 9 defenders in total");
+            } else if (startingDefenders > MAX_DEF) {
+                throw new Error400("Fantasy team with id " + fantasyTeamId + " cannot have more than 5 defenders in the starting lineup");
```

Comment:
if totalPosition > TOTAL_POSITION throw error is good but
if startingPosition > MAX_POSITION should return false, not throw error, because in that case, there might be an option on the bench